tidb-operator panic in `tidbcluster.go`

## Bug Report

**What version of Kubernetes are you using?**
IDC

**What version of TiDB Operator are you using?**
<!-- You can run `kubectl exec -n tidb-admin {tidb-controller-manager-pod-name} -- tidb-controller-manager -V` -->
master
**What storage classes exist in the Kubernetes cluster and what are used for PD/TiKV pods?**
<!-- You can run `kubectl get sc` and `kubectl get pvc -n {tidb-cluster-namespace}` -->

**What's the status of the TiDB cluster pods?**
<!-- You can run `kubectl get po -n {tidb-cluster-namespace} -o wide` -->

**What did you do?**
<!-- If possible, provide a recipe for reproducing the error. How you installed tidb-operator and tidb-cluster. -->
apply tidb-cluster.yaml without tidb fields.
**What did you expect to see?**
apply tidb-cluster.yaml
```
apiVersion: pingcap.com/v1alpha1
kind: TidbCluster
metadata:
  name: test
  namespace: andy
spec:
  version: latest
  timezone: UTC
  enableTLSCluster: false
  pvReclaimPolicy: Delete
  enableAdvertiseAddress: true
  autoFailover: false
  configUpdateStrategy: RollingUpdate
  pd:
    baseImage: hub.xxx.xxx/csn/pd
    version: 5.0-hot-4a98c86
    replicas: 1
    service:
      type: NodePort
    storageClassName: shared-nvme-disks
    limits:
      cpu: 6000m
      memory: 8Gi
    requests:
      cpu: 6000m
      memory: 8Gi
      storage: 200Gi
    config:
      log:
        level: "info"
        file:
          filename: /var/log/pdlog/pd.log
      dashboard:
        enable-telemetry: false
        enable-experimental: true
    storageVolumes:
      - name: log
        storageClassName: shared-sas-disks
        storageSize: "20Gi"
        mountPath: "/var/log/pdlog"
  tikv:
    env:
      - name: TEST_TIKV_TAGS
        value: "test_need_restart"
    baseImage: hub.xxx.xxx/csn/tikv
    version: 5.0-rc-2-1142bc4
    replicas: 3
    storageClassName: nvme-disks
    maxFailoverCount: 0
    evictLeaderTimeout: 1h
    limits:
      cpu: 8000m
      memory: 10Gi
      storage: 50Gi
    requests:
      cpu: 8000m
      memory: 10Gi
      storage: 50Gi
    storageVolumes:
      - name: log
        storageClassName: shared-sas-disks
        storageSize: "20Gi"
        mountPath: "/var/log/tikvlog"

```
**What did you see instead?**
```
 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 685 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x1a9d9c0, 0x2fa4210)
	k8s.io/apimachinery@v0.0.0/pkg/util/runtime/runtime.go:74 +0xa3
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	k8s.io/apimachinery@v0.0.0/pkg/util/runtime/runtime.go:48 +0x82
panic(0x1a9d9c0, 0x2fa4210)
	runtime/panic.go:679 +0x1b2
github.com/pingcap/tidb-operator/pkg/apis/pingcap/v1alpha1.(*TiDBSpec).IsTLSClientEnabled(...)
	github.com/pingcap/tidb-operator@/pkg/apis/pingcap/v1alpha1/tidbcluster.go:576
github.com/pingcap/tidb-operator/pkg/manager/member.getPDConfigMap(0xc0017c2000, 0x0, 0x0, 0x0)
	github.com/pingcap/tidb-operator@/pkg/manager/member/pd_member_manager.go:797 +0xbb
github.com/pingcap/tidb-operator/pkg/manager/member.(*pdMemberManager).syncPDConfigMap(0xc000a4f3c0, 0xc0017c2000, 0x0, 0x0, 0x0, 0xc0010fd220)
	github.com/pingcap/tidb-operator@/pkg/manager/member/pd_member_manager.go:383 +0x47
github.com/pingcap/tidb-operator/pkg/manager/member.(*pdMemberManager).syncPDStatefulSetForTidbCluster(0xc000a4f3c0, 0xc0017c2000, 0x0, 0x0)
	github.com/pingcap/tidb-operator@/pkg/manager/member/pd_member_manager.go:191 +0x54a
github.com/pingcap/tidb-operator/pkg/manager/member.(*pdMemberManager).Sync(0xc000a4f3c0, 0xc0017c2000, 0x0, 0x0)
	github.com/pingcap/tidb-operator@/pkg/manager/member/pd_member_manager.go:83 +0xa9
github.com/pingcap/tidb-operator/pkg/controller/tidbcluster.(*defaultTidbClusterControl).updateTidbCluster(0xc000220d00, 0xc0017c2000, 0xc001415c01, 0x12e729d)
	github.com/pingcap/tidb-operator@/pkg/controller/tidbcluster/tidb_cluster_control.go:172 +0x308
github.com/pingcap/tidb-operator/pkg/controller/tidbcluster.(*defaultTidbClusterControl).UpdateTidbCluster(0xc000220d00, 0xc0017c2000, 0x0, 0x0)
	github.com/pingcap/tidb-operator@/pkg/controller/tidbcluster/tidb_cluster_control.go:106 +0xc8
github.com/pingcap/tidb-operator/pkg/controller/tidbcluster.(*Controller).syncTidbCluster(...)
	github.com/pingcap/tidb-operator@/pkg/controller/tidbcluster/tidb_cluster_controller.go:156
github.com/pingcap/tidb-operator/pkg/controller/tidbcluster.(*Controller).sync(0xc00086b2c0, 0xc000848120, 0x9, 0x0, 0x0)
	github.com/pingcap/tidb-operator@/pkg/controller/tidbcluster/tidb_cluster_controller.go:152 +0x1f0
github.com/pingcap/tidb-operator/pkg/controller/tidbcluster.(*Controller).processNextWorkItem(0xc00086b2c0, 0x1154b00)
	github.com/pingcap/tidb-operator@/pkg/controller/tidbcluster/tidb_cluster_controller.go:119 +0x100
github.com/pingcap/tidb-operator/pkg/controller/tidbcluster.(*Controller).worker(0xc00086b2c0)
	github.com/pingcap/tidb-operator@/pkg/controller/tidbcluster/tidb_cluster_controller.go:107 +0x2b
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc000f9b940)
	k8s.io/apimachinery@v0.0.0/pkg/util/wait/wait.go:152 +0x5e
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000f9b940, 0x3b9aca00, 0x0, 0xc000168101, 0xc0002e2060)
	k8s.io/apimachinery@v0.0.0/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(0xc000f9b940, 0x3b9aca00, 0xc0002e2060)
	k8s.io/apimachinery@v0.0.0/pkg/util/wait/wait.go:88 +0x4d
created by github.com/pingcap/tidb-operator/pkg/controller/tidbcluster.(*Controller).Run
	github.com/pingcap/tidb-operator@/pkg/controller/tidbcluster/tidb_cluster_controller.go:99 +0x1d9
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x168 pc=0x1707c2b]

goroutine 685 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	k8s.io/apimachinery@v0.0.0/pkg/util/runtime/runtime.go:55 +0x105
panic(0x1a9d9c0, 0x2fa4210)
	runtime/panic.go:679 +0x1b2
github.com/pingcap/tidb-operator/pkg/apis/pingcap/v1alpha1.(*TiDBSpec).IsTLSClientEnabled(...)
	github.com/pingcap/tidb-operator@/pkg/apis/pingcap/v1alpha1/tidbcluster.go:576
github.com/pingcap/tidb-operator/pkg/manager/member.getPDConfigMap(0xc0017c2000, 0x0, 0x0, 0x0)
	github.com/pingcap/tidb-operator@/pkg/manager/member/pd_member_manager.go:797 +0xbb
github.com/pingcap/tidb-operator/pkg/manager/member.(*pdMemberManager).syncPDConfigMap(0xc000a4f3c0, 0xc0017c2000, 0x0, 0x0, 0x0, 0xc0010fd220)
	github.com/pingcap/tidb-operator@/pkg/manager/member/pd_member_manager.go:383 +0x47
github.com/pingcap/tidb-operator/pkg/manager/member.(*pdMemberManager).syncPDStatefulSetForTidbCluster(0xc000a4f3c0, 0xc0017c2000, 0x0, 0x0)
	github.com/pingcap/tidb-operator@/pkg/manager/member/pd_member_manager.go:191 +0x54a
github.com/pingcap/tidb-operator/pkg/manager/member.(*pdMemberManager).Sync(0xc000a4f3c0, 0xc0017c2000, 0x0, 0x0)
	github.com/pingcap/tidb-operator@/pkg/manager/member/pd_member_manager.go:83 +0xa9
github.com/pingcap/tidb-operator/pkg/controller/tidbcluster.(*defaultTidbClusterControl).updateTidbCluster(0xc000220d00, 0xc0017c2000, 0xc001415c01, 0x12e729d)
	github.com/pingcap/tidb-operator@/pkg/controller/tidbcluster/tidb_cluster_control.go:172 +0x308
github.com/pingcap/tidb-operator/pkg/controller/tidbcluster.(*defaultTidbClusterControl).UpdateTidbCluster(0xc000220d00, 0xc0017c2000, 0x0, 0x0)
	github.com/pingcap/tidb-operator@/pkg/controller/tidbcluster/tidb_cluster_control.go:106 +0xc8
github.com/pingcap/tidb-operator/pkg/controller/tidbcluster.(*Controller).syncTidbCluster(...)
	github.com/pingcap/tidb-operator@/pkg/controller/tidbcluster/tidb_cluster_controller.go:156
github.com/pingcap/tidb-operator/pkg/controller/tidbcluster.(*Controller).sync(0xc00086b2c0, 0xc000848120, 0x9, 0x0, 0x0)
	github.com/pingcap/tidb-operator@/pkg/controller/tidbcluster/tidb_cluster_controller.go:152 +0x1f0
github.com/pingcap/tidb-operator/pkg/controller/tidbcluster.(*Controller).processNextWorkItem(0xc00086b2c0, 0x1154b00)
	github.com/pingcap/tidb-operator@/pkg/controller/tidbcluster/tidb_cluster_controller.go:119 +0x100
github.com/pingcap/tidb-operator/pkg/controller/tidbcluster.(*Controller).worker(0xc00086b2c0)
	github.com/pingcap/tidb-operator@/pkg/controller/tidbcluster/tidb_cluster_controller.go:107 +0x2b
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc000f9b940)
	k8s.io/apimachinery@v0.0.0/pkg/util/wait/wait.go:152 +0x5e
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000f9b940, 0x3b9aca00, 0x0, 0xc000168101, 0xc0002e2060)
	k8s.io/apimachinery@v0.0.0/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(0xc000f9b940, 0x3b9aca00, 0xc0002e2060)
	k8s.io/apimachinery@v0.0.0/pkg/util/wait/wait.go:88 +0x4d
created by github.com/pingcap/tidb-operator/pkg/controller/tidbcluster.(*Controller).Run
	github.com/pingcap/tidb-operator@/pkg/controller/tidbcluster/tidb_cluster_controller.go:99 +0x1d9

```

--- a/pkg/manager/member/pd_member_manager.go
+++ b/pkg/manager/member/pd_member_manager.go
@@ -565,7 +565,7 @@ func getNewPDSetForTidbCluster(tc *v1alpha1.TidbCluster, cm *corev1.ConfigMap) (
 			})
 		}
 	}
-	if tc.Spec.TiDB.IsTLSClientEnabled() && !tc.SkipTLSWhenConnectTiDB() && clusterVersionGE4 {
+	if tc.Spec.TiDB != nil && tc.Spec.TiDB.IsTLSClientEnabled() && !tc.SkipTLSWhenConnectTiDB() && clusterVersionGE4 {
 		volMounts = append(volMounts, corev1.VolumeMount{
 			Name: "tidb-client-tls", ReadOnly: true, MountPath: tidbClientCertPath,
 		})
@@ -612,7 +612,7 @@ func getNewPDSetForTidbCluster(tc *v1alpha1.TidbCluster, cm *corev1.ConfigMap) (
 			})
 		}
 	}
-	if tc.Spec.TiDB.IsTLSClientEnabled() && !tc.SkipTLSWhenConnectTiDB() && clusterVersionGE4 {
+	if tc.Spec.TiDB != nil && tc.Spec.TiDB.IsTLSClientEnabled() && !tc.SkipTLSWhenConnectTiDB() && clusterVersionGE4 {
 		clientSecretName := util.TiDBClientTLSSecretName(tc.Name)
 		if tc.Spec.PD.TLSClientSecretName != nil {
 			clientSecretName = *tc.Spec.PD.TLSClientSecretName
@@ -806,7 +806,6 @@ func getNewPDSetForTidbCluster(tc *v1alpha1.TidbCluster, cm *corev1.ConfigMap) (
 }
 
 func getPDConfigMap(tc *v1alpha1.TidbCluster) (*corev1.ConfigMap, error) {
-
 	// For backward compatibility, only sync tidb configmap when .tidb.config is non-nil
 	config := tc.Spec.PD.Config
 	if config == nil {
@@ -825,7 +824,7 @@ func getPDConfigMap(tc *v1alpha1.TidbCluster) (*corev1.ConfigMap, error) {
 		config.Set("security.key-path", path.Join(pdClusterCertPath, corev1.TLSPrivateKeyKey))
 	}
 	// Versions below v4.0 do not support Dashboard
-	if tc.Spec.TiDB.IsTLSClientEnabled() && !tc.SkipTLSWhenConnectTiDB() && clusterVersionGE4 {
+	if tc.Spec.TiDB != nil && tc.Spec.TiDB.IsTLSClientEnabled() && !tc.SkipTLSWhenConnectTiDB() && clusterVersionGE4 {
 		config.Set("dashboard.tidb-cacert-path", path.Join(tidbClientCertPath, tlsSecretRootCAKey))
 		config.Set("dashboard.tidb-cert-path", path.Join(tidbClientCertPath, corev1.TLSCertKey))
 		config.Set("dashboard.tidb-key-path", path.Join(tidbClientCertPath, corev1.TLSPrivateKeyKey))

